src/GitVersionExe/ExecCommand.cs

```diff
@@ -3,50 +3,49 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Runtime.InteropServices;
-using GitVersion.BuildServers;
 using GitVersion.Exceptions;
 using GitVersion.OutputFormatters;
 using GitVersion.OutputVariables;
 using GitVersion.Extensions;
 using GitVersion.Extensions.VersionAssemblyInfoResources;
 using GitVersion.Common;
-using GitVersion.Configuration;
 using GitVersion.Logging;
+using Microsoft.Extensions.Options;
 
 namespace GitVersion
 {
-    public class ExecCommand
+    public class ExecCommand : IExecCommand
     {
         private static readonly bool runningOnUnix = !RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+        private readonly IFileSystem fileSystem;
+        private readonly IBuildServerResolver buildServerResolver;
+        private readonly ILog log;
+        private readonly IExecuteCore executeCore;
+        private readonly Arguments arguments;
         public static readonly string BuildTool = GetMsBuildToolPath();
 
-        public void Execute(Arguments arguments, IFileSystem fileSystem, IEnvironment environment, ILog log, IConfigFileLocator configFileLocator)
+        public ExecCommand(IFileSystem fileSystem, IBuildServerResolver buildServerResolver, ILog log, IExecuteCore executeCore, IOptions<Arguments> arguments)
         {
-            log.Info($"Running on {(runningOnUnix ? "Unix" : "Windows")}.");
+            this.fileSystem = fileSystem;
```

Comment:
Perhaps we can add some `null` checks for most of the `public` accessible methods and constructors in the codebase?

```suggestion
            this.fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
```

Reply:
I'll add as a task for this PR, I'll fix it at the end